### PR TITLE
Shor's Algorithm Tutorial

### PR DESCRIPTION
**Pull request by attp**
Adding start of Shor's Algorithm Tutorial. Have covered the theory, welcome suggestions as to the best way to implement it on on the ibmqx devices.

### CODE REVIEW

**Comment by attp** [comment]
Tutorial is now finished, please review and approve.

**Comment by rraymondhp** [comment]
Thanks.
Please also modify the link at index files.

File: index.ipynb [hunk 1]
```diff
@@ -70,7 +70,7 @@
     "  * [Phase estimation](reference/algorithms/iterative_phase_estimation_algorithm.ipynb) - a quantum algorithm to extract eigenvalues of unknown unitary operator.\n",
     "  * [Simon's algorithm](reference/algorithms/simon_algorithm.ipynb) - a quantum algorithm that is proven exponentially efficient that its classical counterpart\n",
     "  * [Grover search](reference/algorithms/grover_algorithm.ipynb) - using Grover search to solve a combinatorial problem\n",
-    "  * Shor's algorithm\n",
+    "  * [Shor's algorithm](shor_algorithm.ipynb) - a quantum algorithm for integer factorisation.\n",
```
Comment (rraymondhp) [inline index.ipynb]: Shouldn't the link be `reference/algorithms/shor_algorithm.ipynb` ?

**Comment by rraymondhp** [comment]
I see.. there can be conflicts with changes of names that have not been merged...

**Pull request merged**